vaadin-combo-box-flow/src/test/java/com/vaadin/flow/component/combobox/dataview/ComboBoxListDataViewTest.java

```diff
@@ -0,0 +1,213 @@
+package com.vaadin.flow.component.combobox.dataview;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.vaadin.flow.component.combobox.ComboBox;
+import com.vaadin.flow.data.provider.AbstractListDataViewListenerTest;
+import com.vaadin.flow.data.provider.DataCommunicator;
+import com.vaadin.flow.data.provider.DataKeyMapper;
+import com.vaadin.flow.data.provider.HasListDataView;
+import com.vaadin.flow.shared.Registration;
+
+public class ComboBoxListDataViewTest extends AbstractListDataViewListenerTest {
+
+    private List<String> items;
+    private ComboBoxListDataView<String> dataView;
+    private ComboBox<String> component;
+
+    @Before
+    public void init() {
+        items = new ArrayList<>(Arrays.asList("first", "middle", "last"));
+        component = new ComboBox<>();
+        dataView = component.setItems(items);
+    }
+
+    @Test
+    public void getItems_noFiltersSet_allItemsObtained() {
+        Stream<String> allItems = dataView.getItems();
+        Assert.assertArrayEquals("Unexpected data set", items.toArray(),
+                allItems.toArray());
+    }
+
+    @Test
+    public void getItems_withTextFilter_filteredItemsObtained() {
+        ComboBox<String> comboBox = getDefaultLocaleComboBox();
+        setComboBoxTextFilter(comboBox, "middle");
+        ComboBoxListDataView<String> dataView = comboBox.setItems(items);
+        Stream<String> allItems = dataView.getItems();
+        Assert.assertArrayEquals("Unexpected data set",
+                new String[] { "middle" }, allItems.toArray());
+    }
+
+    @Test
+    public void getItems_withInMemoryFilter_filteredItemsObtained() {
+        dataView.setFilter("middle"::equals);
+        Stream<String> allItems = dataView.getItems();
+        Assert.assertArrayEquals("Unexpected data set",
+                new String[] { "middle" }, allItems.toArray());
+    }
+
+    @Test
+    public void getItems_withInMemoryAndTextFilter_filteredItemsObtained() {
+        ComboBox<String> comboBox = getDefaultLocaleComboBox();
+        setComboBoxTextFilter(comboBox, "ba");
+        items = Arrays.asList("foo", "bar", "banana");
+        ComboBoxListDataView<String> dataView = comboBox.setItems(items);
+        dataView.setFilter(item -> item.length() == 3);
+        Stream<String> allItems = dataView.getItems();
+        Assert.assertArrayEquals("Unexpected data set",
+                new String[] { "bar" }, allItems.toArray());
```

Comment:
This is partly a nit, but I see that a lot of times your tests only verify that "if a, then b happens", but not after again that "if a is removed, then c happens" which for instance in this case means that you're not testing changing/removing filters work at all. I'm pretty sure that those work, but as an overall comment, I've seen that is often not tested and very often also the source of bugs that users report, which could have been caught with very little testing effort.

Reply:
Good catch, will improve that

Reply:
Done